debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDatabaseHistoryIT.java

```diff
@@ -0,0 +1,153 @@
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import java.nio.file.Path;
+import java.sql.SQLException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.doc.FixFor;
+import io.debezium.embedded.AbstractConnectorTest;
+import io.debezium.util.Testing;
+
+/**
+*
+* The test to verify whether DDL is stored correctly in database history.
+*
+* @author Jiri Pechanec
+*/
+public class MySqlDatabaseHistoryIT extends AbstractConnectorTest {
+
+    private static final Path DB_HISTORY_PATH = Testing.Files.createTestingPath("file-db-history-json.txt")
+            .toAbsolutePath();
+
+    private static final int TABLE_COUNT = 2;
+
+    private UniqueDatabase DATABASE;
+
+    private Configuration config;
+
+    @Before
+    public void beforeEach() {
+        stopConnector();
+        DATABASE = new UniqueDatabase("history", "history-dbz")
+                .withDbHistoryPath(DB_HISTORY_PATH);
+        DATABASE.createAndInitialize();
+
+        initializeConnectorTestFramework();
+        Testing.Files.delete(DB_HISTORY_PATH);
+    }
+
+    @After
+    public void afterEach() {
+        try {
+            stopConnector();
+        }
+        finally {
+            Testing.Files.delete(DB_HISTORY_PATH);
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-3485")
+    public void shouldUseQuotedNameInDrop() throws SQLException, InterruptedException {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.SCHEMA_ONLY)
+                .build();
+
+        // Start the connector ...
+        start(MySqlConnector.class, config);
+
+        Testing.Print.enable();
+        // SET + USE + Drop DB + create DB + CREATE/DROP for each table
+        SourceRecords records = consumeRecordsByTopic(1 + 1 + 1 + 1 + TABLE_COUNT * 2);
+
+        stopConnector();
+
+        start(MySqlConnector.class, config);
+        assertConnectorIsRunning();
+        stopConnector();
+    }
+
+    @Test
+    @FixFor("DBZ-3399")
+    public void shouldStoreSingleRename() throws SQLException, InterruptedException {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.SCHEMA_ONLY)
+                .build();
+
+        // Start the connector ...
+        start(MySqlConnector.class, config);
+
+        Testing.Print.enable();
+        // SET + USE + Drop DB + create DB + CREATE/DROP for each table
+        SourceRecords records = consumeRecordsByTopic(1 + 1 + 1 + 1 + TABLE_COUNT * 2);
+        try (MySqlTestConnection connection = MySqlTestConnection.forTestDatabase(DATABASE.getDatabaseName())) {
+            connection.execute("RENAME TABLE `t-1` TO `new-t-1`");
+        }
+        records = consumeRecordsByTopic(1);
```

Comment:
How about adding some actual assertion on the event and its attributes instead of solely counting? Same for the other test methods.

Reply:
@gunnarmorling Added